ecs-agent/tmds/utils/netconfig/netconfig_linux.go

```
@@ -69,3 +72,32 @@ func DefaultNetInterfaceName(netlinkClient netlinkwrapper.NetLink) (string, erro
 	}
 	return "", nil
 }
+
+// GetInterfaceGlobalIPAddresses returns all global unicast IP addresses (both IPv4 and IPv6)
+// assigned to the given network interface. It excludes link-local, loopback, multicast,
+// and unspecified addresses. Returns an empty list if no global unicast addresses are found,
+// or an error if the interface cannot be accessed.
+func GetInterfaceGlobalIPAddresses(nw netwrapper.Net, ifaceName string) ([]string, error) {
+	iface, err := nw.InterfaceByName(ifaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	allAddrs, err := nw.Addrs(iface)
+	if err != nil {
+		return nil, err
+	}
+
+	ipAddrs := make([]string, 0)
+	for _, addr := range allAddrs {
+		ipNet, ok := addr.(*net.IPNet)
+		if !ok {
+			continue
```

are we going to swallow this error? Do we want to log something?

---

It's okay because we need IP addresses only.